tests/test_resources.py

```diff
@@ -0,0 +1,248 @@
+        self.token = resp.get('token')
+        return self.token
+
+    def post_bucketlist(self):
+        self.app.post('/api/v1/bucketlists/',
+                      data=self.bucketlist, headers={'Token': self.token})
+
+    def test_bucketlist_item(self):
+        self.reg_user()
+        self.login_user()
+        self.post_bucketlist()
+        #
+        req = {'name': 'Learn Python'}
+        res = self.app.post('/api/v1/bucketlists/1/items/1',
+                            data=req, headers={'Token': self.token})
+        assert res.status_code == 405
+        res = self.app.post('/api/v1/bucketlists/2/items/',
+                            data=req, headers={'Token': self.token})
+        assert res.status_code == 404
+        req = {'name': '          learn'}
+        res = self.app.post('/api/v1/bucketlists/1/items/',
+                            data=req, headers={'Token': self.token})
+        assert res.status_code == 400
+        req = {'name': 'Learn Python'}
+        res = self.app.post('/api/v1/bucketlists/1/items/',
+                            data=req, headers={'Token': self.token})
+        assert res.status_code == 201
+        res = self.app.post('/api/v1/bucketlists/1/items/',
+                            data=req, headers={'Token': self.token})
+        assert res.status_code == 400
+        assert 'item name already exists' in (
+            json.loads(res.data)).get('message')
+        #
+        req = {'name': 'Learn JavaScript'}
+        res = self.app.put('/api/v1/bucketlists/1/items/',
+                           data=req, headers={'Token': self.token})
+        assert res.status_code == 405
+        res = self.app.put('/api/v1/bucketlists/2/items/1',
+                           data=req, headers={'Token': self.token})
+        assert res.status_code == 404
+        req = {'name': 'Learn          '}
+        res = self.app.put('/api/v1/bucketlists/1/items/1',
+                           data=req, headers={'Token': self.token})
+        assert res.status_code == 400
+        req = {'name': 'Learn JavaScript'}
+        res = self.app.put('/api/v1/bucketlists/1/items/1',
+                           data=req, headers={'Token': self.token})
+        res.status_code == 201
+        req = {'done': 'True'}
+        res = self.app.put('/api/v1/bucketlists/1/items/1',
+                           data=req, headers={'Token': self.token})
+        assert res.status_code == 201
+        res = self.app.delete('/api/v1/bucketlists/1/items/',
+                              headers={'Token': self.token})
+        assert res.status_code == 405
+        res = self.app.delete('/api/v1/bucketlists/1/items/2',
+                              headers={'Token': self.token})
+        assert res.status_code == 404
+        res = self.app.delete('/api/v1/bucketlists/1/items/1',
+                              headers={'Token': self.token})
+        assert res.status_code == 200
+
```

Comment:
This is a crime :). You shouldn't be doing this. Break this out into methods that define what each block is testing.